qt/TorrentFilter.cc

```diff
@@ -75,6 +75,31 @@ void TorrentFilter::refilter()
 ****
 ***/
 
+namespace
+{
+int compareState(Torrent const* left, Torrent const* right)
+{
+    int val = tr_compare_3way(left->hasError(), right->hasError());
+
+    if (val == 0)
+    {
+        val = tr_compare_3way(left->isFinished(), right->isFinished());
+    }
```

Comment:
_(nit)_

:+1: on extracting this to its own method.

The `if (val == 0) ...` code isn't new to this PR, and the previous code is mostly a holdover from the pre-C++-17 days. We generally try to use `if-` variable declarations for a smaller variable scope + more `return` calls:

```c++
if (auto const val = tr_compare_3way(left->hasError(), right->hasError()); val != 0)
{
    return val;
}

if (auto const val = tr_compare_3way(left->isFinished(), right->isFinished()); val != 0)
{
    return val;
}

...
```

Reply:
c085cdc6e209cfbeb5c507906944892d9bbbd0b5